gnovm/stdlibs/std/emit_event.go

```diff
@@ -0,0 +1,77 @@
+package std
+
+// ref: https://github.com/gnolang/gno/pull/853
+
+import (
+	"encoding/json"
+	"errors"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+)
+
+func X_emit(m *gno.Machine, typ string, attrs []string) {
+	eventAttrs, err := attrKeysAndValues(attrs)
+	if err != nil {
+		m.Panic(typedString(err.Error()))
+	}
+
+	pkgPath := CurrentRealmPath(m)
+	fnIdent := getPrevFunctionNameFromTarget(m, "Emit")
+	timestamp := getTimestamp(m)
+
+	evtstr := NewGnoEventString(typ, pkgPath, fnIdent, timestamp, eventAttrs...)
+	ctx := m.Context.(ExecContext)
+	ctx.EventLogger.EmitEvent(evtstr)
+}
+
+func attrKeysAndValues(attrs []string) ([]gnoEventAttribute, error) {
+	attrLen := len(attrs)
+	if attrLen%2 != 0 {
+		return nil, errors.New("odd number of attributes. cannot create key-value pairs")
+	}
+	eventAttrs := make([]gnoEventAttribute, attrLen/2)
+	for i := 0; i < attrLen-1; i += 2 {
+		eventAttrs[i/2] = gnoEventAttribute{
+			Key:   attrs[i],
+			Value: attrs[i+1],
+		}
+	}
+	return eventAttrs, nil
+}
+
+type temp gnoEvent
```

Comment:
What is the purpose of this? (If there is one, please provide a comment or suggest a clearer name.)

Reply:
this is a temporary type introduced to prevent possible overflow during marshaling. but the exaplanation was insufficient. I will add comments to this part.

https://github.com/gnolang/gno/pull/1653#discussion_r1571006721

Reply:
it was a temporary type to prevent overflow when using event type as a JSON key previously. But now there is no need to pu the type inside the JSON, so it has been removed now.